### ISSUE TITLE
LDAP Login fails, some chats are invisible

### ISSUE THREAD

**Issue by Isotop7** [RocketChat/Rocket.Chat#7801]
### Description:

After upgrading to 0.57.2 we cant login with our Active Directory accounts. We only are able to log in via Gitlab oAuth which also uses our AD as backend.

Also some private chats are not loaded when the page is loaded.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.57.3
* Operating System: Official rc docker container image
* Deployment Method(snap/docker/tar/etc): docker container on rancher os
* Number of Running Instances: 1
* DB Replicaset Oplog: ?
* Node Version: Official rc docker container image

### Steps to Reproduce:

1. Enable ldap users
2. Update to 0.57.2
3. Try to login with ad account
4. Login over oAuth
5. Open private chat page

### Expected behavior:

User logged in, chats loaded

### Actual behavior:

User only logs in over oAuth, private chat windows are not loaded

### Relevant logs:
Login:

```
18.8.2017 18:03:30LDAP ➔ Connection.info LDAP connected
18.8.2017 18:03:30LDAP ➔ Bind.info Binding admin user <ldap bind dn>
18.8.2017 18:03:30LDAP ➔ Search.info Searching user hendrik.roeder@infomotion.de
18.8.2017 18:03:30LDAP ➔ Search.debug searchOptions { filter: '(&(objectCategory=person)(objectclass=user)(|(mail=<mail>)(sAMAccountName=<samaccountname>)))',
18.8.2017 18:03:30  scope: 'sub' }
18.8.2017 18:03:30LDAP ➔ Search.debug domain_base <domain>
18.8.2017 18:03:30LDAP ➔ Search.info Search result count 1
18.8.2017 18:03:30LDAP ➔ Search.debug Search result [
18.8.2017 18:03:30  {
18.8.2017 18:03:30    "messageID": 2,
18.8.2017 18:03:30    "protocolOp": "SearchEntry",
18.8.2017 18:03:30    "objectName": "<object name>",
18.8.2017 18:03:30    "attributes": [
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "objectClass",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "top",
18.8.2017 18:03:30          "person",
18.8.2017 18:03:30          "organizationalPerson",
18.8.2017 18:03:30          "user"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "cn",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<cn>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "sn",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<sn>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "givenName",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<given name>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "initials",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "HERÖ"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "distinguishedName",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<user dn>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "instanceType",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "4"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "whenCreated",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "20160407142942.0Z"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "whenChanged",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "20170809083048.0Z"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "displayName",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<display name>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "uSNCreated",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "26908"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "memberOf",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          <groups>
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "garbageCollPeriod",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "1209600"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "name",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<name>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "objectGUID",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "5��Ϙ'\u0006G�>\u0006��\\;Q"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "userAccountControl",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "512"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "badPwdCount",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "0"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "codePage",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "0"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "countryCode",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "0"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "homeDrive",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "I:"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "badPasswordTime",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "131466532292301581"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "lastLogoff",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "0"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "lastLogon",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "131475404584128293"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "pwdLastSet",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "131420867765164915"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "primaryGroupID",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "513"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "objectSid",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "\u0001\u0005\u0000\u0000\u0000\u0000\u0000\u0005\u0015\u0000\u0000\u0000��yq��\t�\u0004䁨�\u0015\u0000\u0000"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "adminCount",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "1"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "accountExpires",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "9223372036854775807"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "logonCount",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "2599"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "sAMAccountName",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<samaccountname>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "userPrincipalName",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<pn>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "objectCategory",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<domain scheme cn>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "mSMQSignCertificates",
18.8.2017 18:03:30        "vals": [
Unknown Date
Invalid Date Invalid Date          "\u0001\u0000\u0000\u0000��כ˵�.]G9Mv��L��*('�K�Z\u0013q`%ǵd\u0003\u0000\u00000�\u0003`0�\u0002H�\u0003\u0002\u0001\u0002\u0002\u0004��ZU0\r\u0006\t*�H��\r\u0001\u0001\u0005\u0005\u00000r1\u00110\u000f\u0006\u0003U\u0004\u0007\u001e\b\u0000M\u0000S\u0000M\u0000Q1\u000b0\t\u0006\u0003U\u0004\n\u001e\u0002\u0000-1\u000b0\t\u0006\u0003U\u0004\u000b\u001e\u0002\u0000-1C0A\u0006\u0003U\u0004\u0003\u001e:\u0000I\u0000N\u0000F\u0000O\u0000M\u0000O\u0000T\u0000I\u0000O\u0000N\u0000\\\u0000h\u0000r\u0000o\u0000e\u0000d\u0000e\u0000r\u0000,\u0000 \u0000t\u0000k\u0000-\u0000i\u0000m\u0000-\u0000s\u0000t\u0000u0\u001e\u0017\r170403091935Z\u0017\r250403091935Z0r1\u00110\u000f\u0006\u0003U\u0004\u0007\u001e\b\u0000M\u0000S\u0000M\u0000Q1\u000b0\t\u0006\u0003U\u0004\n\u001e\u0002\u0000-1\u000b0\t\u0006\u0003U\u0004\u000b\u001e\u0002\u0000-1C0A\u0006\u0003U\u0004\u0003\u001e:\u0000I\u0000N\u0000F\u0000O\u0000M\u0000O\u0000T\u0000I\u0000O\u0000N\u0000\\\u0000h\u0000r\u0000o\u0000e\u0000d\u0000e\u0000r\u0000,\u0000 \u0000t\u0000k\u0000-\u0000i\u0000m\u0000-\u0000s\u0000t\u0000u0�\u0001\"0\r\u0006\t*�H��\r\u0001\u0001\u0001\u0005\u0000\u0003�\u0001\u000f\u00000�\u0001\n\u0002�\u0001\u0001\u0000�2\u0004\u001fn7��\u0015�@���7 �2���(+\u0018�r�&b��S\"oR��$\u001c�\r�Ȃ�6��\u0003��[i��%���[��\u0000�M\\�EB��}X���_��\u001e�2h�!Ƭ8���@��<�[@�F\"�`���TY��\u001e��\b�^�����)׼�bʚ(��'3z\u0010~R�]s'�\u00117-�;�j\u0010���T�+\u0019����4�?M}\f��j�!D��q\u000e�a@�6`�W<υ�\u000e(f놕�i�\u0002�d��Z\u0002��x�9\u0005u\u0002��,�p�\"��l�r\nX@\u00017\u0003��S}%��Y$j�]eh�7���\u000bWd�\u0002\u0003\u0001\u0000\u00010\r\u0006\t*�H��\r\u0001\u0001\u0004\u0005\u0000\u0003�\u0001\u0001\u0000\u000f���㦷�f\u000e\\��]��N��)\u001f\u001b��s\u0012�\u0004 ��f�\u0003�(B3���lmG�s'^_1\u0015��j���\u0017\u0014*nV6�\u001ee5m\u0018xF40q������K�Ӛ����4K?�\u0016*_\u000e�)6�[4�ġ�aeR\u0019�-�pq%U00\u0004���\u000b���\u001f����g��Rz�\u001cg!�\u0002����>�n\u001c>:k�s ���S�E\u0002\u0013�E�BGI\u000e����\u000f@c�g�\u0003�\u0018\u001a^�iG\u0017��\u0017+=DJ�\u001b�=)\u0004D���e\u0018\u000e]�,6�\nv��������۔R1~8\u0006\u000b��\u0000�\u0014������v@h\u001d�FF�E"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "mSMQDigests",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "��כ˵�.]G9Mv��L"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "mail",
18.8.2017 18:03:30        "vals": [
18.8.2017 18:03:30          "<mail>"
18.8.2017 18:03:30        ]
18.8.2017 18:03:30      },
...
18.8.2017 18:03:30      {
18.8.2017 18:03:30        "type": "thumbnailPhoto",
18.8.2017 18:03:30        "vals": [
```

When  loading chat window:

```
18.8.2017 18:10:02Meteor ➔ publish roles -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02BaseCache ➔ debug rocketchat_roles Falling back _publishCursor to model with query: undefined
18.8.2017 18:10:02Meteor ➔ publish meteor_autoupdate_clientVersions -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02Meteor ➔ publish stream-notify-all -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'updateCustomSound', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-all -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'deleteCustomSound', '1': false }
18.8.2017 18:10:02Meteor ➔ publish activeUsers -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02BaseCache ➔ debug users Falling back _publishCursor to model with query: { username: { '$exists': 1 },
18.8.2017 18:10:02  status: { '$in': [ 'online', 'away', 'busy' ] } }
18.8.2017 18:10:02Meteor ➔ publish userData -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02BaseCache ➔ debug users Falling back _publishCursor to model with query: uSTQo4QJMNoxuh2Zd
18.8.2017 18:10:02Meteor ➔ publish stream-notify-all -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'public-settings-changed', '1': false }
18.8.2017 18:10:02Meteor ➔ publish null -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'roles-change', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'updateEmojiCustom', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'deleteEmojiCustom', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'Users:NameChanged', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'updateAvatar', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'permissions-changed', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-logged -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'private-settings-changed', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-user -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'uSTQo4QJMNoxuh2Zd/message', '1': false }
18.8.2017 18:10:02Meteor ➔ method getUserRoles -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02Meteor ➔ publish stream-notify-user -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'uSTQo4QJMNoxuh2Zd/otr', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-user -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'uSTQo4QJMNoxuh2Zd/webrtc', '1': false }
18.8.2017 18:10:02Meteor ➔ publish stream-notify-user -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'uSTQo4QJMNoxuh2Zd/notification', '1': false }
18.8.2017 18:10:02Meteor ➔ publish userData -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02BaseCache ➔ debug users Falling back _publishCursor to model with query: uSTQo4QJMNoxuh2Zd
18.8.2017 18:10:02Meteor ➔ publish activeUsers -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02BaseCache ➔ debug users Falling back _publishCursor to model with query: { username: { '$exists': 1 },
18.8.2017 18:10:02  status: { '$in': [ 'online', 'away', 'busy' ] } }
18.8.2017 18:10:02Meteor ➔ method rooms/get -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02Meteor ➔ publish stream-notify-user -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'uSTQo4QJMNoxuh2Zd/rooms-changed', '1': false }
18.8.2017 18:10:02Meteor ➔ method subscriptions/get -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02Meteor ➔ publish stream-notify-user -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': 'uSTQo4QJMNoxuh2Zd/subscriptions-changed', '1': false }
18.8.2017 18:10:02Meteor ➔ method private-settings/get -> userId: uSTQo4QJMNoxuh2Zd , arguments:  {}
18.8.2017 18:10:02Meteor ➔ method spotlight -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': '',
18.8.2017 18:10:02  '1':
18.8.2017 18:10:02   [ { _id: 'uSTQo4QJMNoxuh2Zd',
18.8.2017 18:10:02       services: [Object],
18.8.2017 18:10:02       username: '<usernam>',
18.8.2017 18:10:02       status: 'offline',
18.8.2017 18:10:02       active: true,
18.8.2017 18:10:02       roles: [Object],
18.8.2017 18:10:02       name: '<name>',
18.8.2017 18:10:02       statusConnection: 'online',
18.8.2017 18:10:02       utcOffset: 2,
18.8.2017 18:10:02       avatarOrigin: 'ldap',
18.8.2017 18:10:02       statusLivechat: 'available',
18.8.2017 18:10:02       statusDefault: 'offline',
18.8.2017 18:10:02       settings: [Object],
18.8.2017 18:10:02       language: 'de',
18.8.2017 18:10:02       requirePasswordChange: true,
18.8.2017 18:10:02       requirePasswordChangeReason: 'This_email_has_already_been_used_and_has_not_been_verified__Please_change_your_password',
18.8.2017 18:10:02       emails: [Object] }, <users> ],
18.8.2017 18:10:02  '2': { users: true, rooms: true } }
18.8.2017 18:10:03Meteor ➔ method setUserPassword -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': '' }
18.8.2017 18:10:04Meteor ➔ method spotlight -> userId: uSTQo4QJMNoxuh2Zd , arguments:  { '0': '',
18.8.2017 18:10:04  '1':
18.8.2017 18:10:04   [ { _id: 'uSTQo4QJMNoxuh2Zd',
18.8.2017 18:10:04       services: [Object],
18.8.2017 18:10:04       username: '<username>',
18.8.2017 18:10:04       status: 'offline',
18.8.2017 18:10:04       active: true,
18.8.2017 18:10:04       roles: [Object],
18.8.2017 18:10:04       name: '<name>',
18.8.2017 18:10:04       statusConnection: 'online',
18.8.2017 18:10:04       utcOffset: 2,
18.8.2017 18:10:04       avatarOrigin: 'ldap',
18.8.2017 18:10:04       statusLivechat: 'available',
18.8.2017 18:10:04       statusDefault: 'offline',
18.8.2017 18:10:04       settings: [Object],
18.8.2017 18:10:04       language: 'de',
18.8.2017 18:10:04       emails: [Object] },  [ <users> ],
18.8.2017 18:10:04  '2': { users: true, rooms: true } }
```

**Comment by Isotop7** [comment]
We still are not able to use Rocket.Chat with AD authentication. We recently updated to version `0.58.2` but the error still occurs.

**Comment by Isotop7** [comment]
This was fixed in `0.59.3`.

Thank you!
